computing.md

```diff
@@ -0,0 +1,142 @@
+# Setting up your computer
+
+Most projects developed in the lab will require you to have 
+git, a terminal, and Python properly setup on your computer.
+Below are some instructions on how to do this.
+
+## Git and a terminal
+
+[Git](https://git-scm.com/) is what we use to collaborate on 
+projects, track a history of changes, and backup to the cloud
+(GitHub). 
+
+A terminal with a decent shell (e.g., bash) is the primary
+interface for using git and other command line utilities
+(LaTeX compilers, make, etc).
+
+### Windows
+
+You can get both git and a bash-enabled terminal with "Git Bash".
+Follow the [Software Carpentry setup instructions](https://carpentries.github.io/workshop-template/#shell)
+to get going.
+
+### Linux
+
+You should already have a terminal with bash (look for the "terminal" app).
+Git is often already installed as well or you can install it with
+your distributions package manager.
+
+On Ubuntu:
+
+```bash
+sudo apt-get install git
+```
+
+## Python
+
+> **DO NOT** download Python from https://www.python.org!
+
+The best way to get setup with Python for your project is 
+by getting a Python distribution that has the `conda` or `mamba`
+package managers. Anaconda itself (which you likely have from
+one of Leo's classes) is good for some cases but for your 
+project work it will likely cause more harm than good because
+it comes with too many libraries that we won't need (and is thus 
+a huge pain to update).
+
+The recommended distribution to install for Windows, Linux, and Mac 
+is [Mambaforge](https://github.com/conda-forge/miniforge#mambaforge).
+This is a minimalist version of Anaconda made by the 
+[conda-forge](https://conda-forge.org/) team. It comes only with Python
+and `mamba` (which we use to install other libraries).
+
+Follow the instructions below to download and setup Mambaforge.
+
+### Windows
+
+Download the [latest Mambaforge](https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Windows-x86_64.exe) 
+and run the installer. 
+
+This should create a `mambaforge` folder in your home directory
+(open Git Bash and run `cd` to get to your home).
+
+Now we need to make Git Bash aware of Mambaforge so that
+you have access to the `mamba` package manager.
+
+> Run the commands below on Git Bash.
+
+Run this command to append the initialization code your
+`.bashrc` file:
+
+```bash
+cat >> .bashrc <<EOF 
+# Mambaforge initialization code
+source ~/mambaforge/etc/profile.d/conda.sh
```

Comment:
```suggestion
source "$HOME/mambaforge/etc/profile.d/conda.sh"
```

Reply:
Maybe changing it to this will work. 

@indiauppal did the installer put things in a `mambaforge` folder in your home?

Reply:
That didn't work either. The path to conda.sh is correct.

Reply:
Wait, can you check if the code is actually appended to the `.bashrc` file? Do you get any errors when you open Git Bash?

Reply:
Stepping in into this thread: maybe it's wiser to use `./conda.sh init bash` to automatically set up bash to source `conda.sh`.

https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#conda-init

BTW, `conda init` works with other shells, e.g. `conda init zsh`, which might be helpful for Mac users (I think they are shipping zsh by default nowdays).

Reply:
I've tried running python and mamba today and its worked. I don't know why its working now and not the other day. Also, every time I open git bash the following code:
 ```
#Mambaforge initialization code
source ~/mambaforge/etc/profile.d/conda.sh
conda activate
```
keeps being added to the end of my .bashrc file. Is this suppose to happen?

Reply:
source ~/AppData/Local/mambaforge/etc/profile.d/conda.sh